sys/include/net/ppp/hdr.h

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (C) 2016 José Ignacio Alamos <jialamos@uc.cl>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_ppp Header
+ * @ingroup     net_ppp_hdr
+ * @brief       PPP header abstraction type and helper functions
+ * @{
+ *
+ * @file
+ * @brief   General definitions for PPP header and their helper functions
+ *
+ * @author  José Ignacio Alamos
+ */
+
+#ifndef PPP_HDR_H_
+#define PPP_HDR_H_
+
+#include <inttypes.h>
+
+#include "byteorder.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+/**
+ * @brief   Header of a PPP packet
+ * @details A PPP packet is transmited as a payload of an HDLC packet. PPP packets only carry information about control protocol
+ * of a PPP stack (Link Control Protocol, IP Control Protocol, etc). IP packets encapsulated in HDLC frame are not
+ * considered PPP packet.
+ *
+ * The format of PPP header plus payload is:
+ *
+ *
+ *  0                   1                   2                   3
+ *  0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * |     Code      |  Identifier   |            Length             |
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * |    Payload ...
+ * +-+-+-+-+
+ *
```

Comment:
Can you maybe put a reference to the RFC or something here? See [`ipv6_hdr_t`](https://github.com/RIOT-OS/RIOT/blob/master/sys/include/net/ipv6.h#L38-L40) for example.